spectral_cube/dask_spectral_cube.py

```diff
@@ -80,7 +80,7 @@ def wrapper(self, *args, **kwargs):
                 raise ImportError("saving the cube to a temporary directory "
                                   "requires the zarr and fsspec packages to "
                                   "be installed.")
-            filename = tempfile.mktemp()
+            filenum, filename = tempfile.mkstemp()
             with dask.config.set(**cube._scheduler_kwargs):
                 cube._data.to_zarr(filename)
             cube._data = da.from_zarr(filename)
```

Comment:
```suggestion
            tmpdir = tempfile.TemporaryDirectory()
            with dask.config.set(**cube._scheduler_kwargs):
                cube._data.to_zarr(tmpdir)
            cube._data = da.from_zarr(tmpdir)
            tmpdir.cleanup()
```
Without the `.cleanup()` the test raises a warning over the leftover directory; it's not quite clear to me if the saved data in tmp_dir are never to be accessed again...